src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/Rule.cs

```diff
@@ -16,5 +16,10 @@ public abstract class Rule
         /// </summary>
         /// <param name="context">The context that the <see cref="IRuleRunner"/> creates holding callbacks to get the differences.</param>
         public abstract void Initialize(RuleRunnerContext context);
+
+        /// <summary>
+        /// Flag that rules should use to determine which mode the differences should be calculated.
+        /// </summary>
+        public bool StrictMode { get; set; }
```

Comment:
I thought about that, and the rules can cache the value when Initialize is invoked. However that would duplicate work across all rules to store the value, that's why I thought this would share the logic better, but I think I also like the other approach.

Whatever you think is better I'll go with that.

Reply:
What's going to happen when you add more settings?  Say filters, the option to check internals, etc.  Imagine how many things might need to change?

Here's a thought.  What if there was a different method on Rule that was not virtual and was responsible for the base setup.  It could set whatever fields you want and those would be readonly so that they couldn't be changed after initialization.  That method could then call Initialize on the derived types?

If you imagine a lot of settings, it could make sense to have a settings class.

The way you have it is fine too, it just struck me that I wouldn't expect that setting to be something changeable after a rule was created.

Reply:
> Say filters, the option to check internals, etc. Imagine how many things might need to change?

Filterings happen at another layer, those are stored in the settings and the mappers filter out when they expand their own tree, so the rules will not see things that are filtered. i.e: https://github.com/dotnet/sdk/blob/51a227376de31d4b441747a15a0d08d6c7b05b01/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/Mappers/NamespaceMapper.cs#L92

> If you imagine a lot of settings, it could make sense to have a settings class.

Yes for rule settings if we start seeing more settings added I think creating a `RuleSettings` class would make sense. 

> Here's a thought. What if there was a different method on Rule that was not virtual and was responsible for the base setup. It could set whatever fields you want and those would be readonly so that they couldn't be changed after initialization. That method could then call Initialize on the derived types?

Yes, that is a good idea, I'll go with that 😄 -- thanks for your suggestions.

Reply:
I went ahead and created `RuleSettings` class as it seems reasonable to use that for all the settings rules should care about.

Reply:
Fixed; Let me know if you have any feedback.